share/mk/bsd.prog.mk

```diff
@@ -48,11 +48,15 @@ LDFLAGS+= -Wl,-znow
 .if ${MK_PIE} != "no"
 # Static PIE is not yet supported/tested.
 .if !defined(NO_SHARED) || ${NO_SHARED:tl} == "no"
+# PIE does not work for libcheri
+.if (defined(WANT_CHERI) && ${WANT_CHERI} == "sandbox") || \
+	(defined(LIBADD) && !empty(LIBADD:Mcheri))
```

Comment:
WANT_CHERI should be phased out. Checking LIBADD seems wrong; what about if a program indirectly links against libcheri?

Reply:
Though I see now this is just inherited from below...

Reply:
Yeah, I'd actually rather just kill this and explicitly set MK_PIE=no in relevant Makefiles, but I'm not sure how many places I'd have to track down to make that work.

Reply:
I've killed some of this (the WANT_CHERI bits) as it shouldn't be needed.

For libcheri we could explicitly set MK_PIE=no in the Makefiles of everything that links against libcheri which would let us remove the rest of this.  That seems a bit error-prone to do, but there are only about 10 places or so to set.

Reply:
I think I prefer disabling pie for the libcheri bits. Shouldn't be that many and I think most might already be static so don't use pie?